### ISSUE TITLE
Running validation more often than at the end of the epoch.

### ISSUE THREAD

**Issue by ayl** [fchollet/keras#5612]
Hello,

I have used offline generation to make a large number of training examples (~millions) and I would like to evaluate the fit with the validation set more frequently than just at the very end of the epoch. Is there a good way to do this within keras's current methods? I could not see an option that would let me set a the number of batches at which the validation accuracy could be tested. 

Thank you

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [X] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [X] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

**Comment by unrealwill** [comment]
Hello,

I'm not aware of such functionality. 
You can probably do a quick "hack" by using fit_generator to create smaller epochs. 
That's usually the way to handle large datasets because large datasets can't usually be stored in memory.

**Comment by gspell** [comment]
I think it would be great to renew this conversation.  Evaluating at most once every epoch seems too limiting for an API as powerful as Keras.
